pkg/controller.v1alpha3/suggestion/suggestionclient/algorithm_settings.go

```diff
@@ -0,0 +1,53 @@
+package suggestionclient
+
+import (
+	common "github.com/kubeflow/katib/pkg/apis/controller/common/v1alpha3"
+	experimentsv1alpha3 "github.com/kubeflow/katib/pkg/apis/controller/experiments/v1alpha3"
+	suggestionsv1alpha3 "github.com/kubeflow/katib/pkg/apis/controller/suggestions/v1alpha3"
+	suggestionapi "github.com/kubeflow/katib/pkg/apis/manager/v1alpha3"
+)
+
+// appendAlgorithmSettingsFromSuggestion appends the algorithm settings
+// in suggestion to Experiment.
+// Algorithm settings in suggestion will overwrite the settings in experiment.
+func appendAlgorithmSettingsFromSuggestion(experiment *experimentsv1alpha3.Experiment, algoSettingsInSuggestion *common.AlgorithmSpec) {
+	algoSettingsInExperiment := experiment.Spec.Algorithm
+	for _, setting := range algoSettingsInSuggestion.AlgorithmSettings {
```

Comment:
Will this be null when algorithmSetting is not set?

Reply:
Will make sure it.